Source/Core/Core/Debugger/PPCDebugInterface.cpp

```diff
@@ -324,6 +325,40 @@ std::string PPCDebugInterface::GetDescription(unsigned int address)
   return g_symbolDB.GetDescription(address);
 }
 
+u32 PPCDebugInterface::GetMemoryAddressFromInstruction(std::string instruction)
+{
+  int offs = 0;
+  u32 roffs = 0;
+  std::regex re(",\\W*(-?0[xX]?[0-9a-fA-F]*|r\\d+)[^r^s]*.(p|toc|\\d+)");
+  std::smatch match;
+  std::regex_search(instruction, match, re);
+
+  // Output: match.str(1): either -+0xNNNN, 0, or rNN. Check for r later to see if a gpr needs to be
+  // loaded. match.str(2): will either be p, toc, or NN. Always a gpr.
+  if (match.str(1).find("r") != std::string::npos)
```

Comment:
The result of `match.str(1)` should be stored to its own variable, otherwise this is redundantly creating the same std::string instance inside the if statements. `"r"` can also be turned into a character literal to allow library implementations to potentially take a cheaper find algorithm (it's quicker to find a single character than to find a string).